warn-stealers.lua

```diff
@@ -0,0 +1,107 @@
+-- Script to warn when creatures that may steal food become visible
+--@ enable = true
+--[====[
+warn-stealers
+=============
+Will make a zoomable announcement whenever a creature that can eat food, guzzle drinks, or steal items enters the map and moves into a revealed location.
+Takes ``start`` or ``stop`` as parameters.
+]====]
+
+local persistTable = require("persist-table")
+local eventful = require("plugins.eventful")
+local repeatUtil = require("repeat-util")
+
+local eventfulKey = "warn-stealers"
+local numTicksBetweenChecks = 100
+
+function gamemodeCheck()
+    if df.global.gamemode ~= df.game_mode.DWARF then
```

Comment:
Prefer putting the smaller code block inside the conditional. In this case, try returning true if the game mode is dwarf. Then the rest of the function can lose an indent level.